### PR TITLE
tests: Add e2e tests for Grafana data sources

### PR DESCRIPTION
**Pull request by aarnq**
> [!warning]
> **This is public repository, ensure not to disclose:**
> - [x] personal data beyond what is necessary for interacting with this pull request
> - [x] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### What does this PR do / why do we need this PR?

There in an issue that the container cannot access the users gpg keyring so I'll look into some solution for that.
Also I will add some bats tests to check the loader container for errors.

- Fixes https://github.com/elastisys/ck8s-ops/issues/1845

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [x] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy

### CODE REVIEW

File: tests/end-to-end/grafana-dev.cy.js [hunk 1]
```diff
@@ -23,4 +23,30 @@ describe("grafana-dev integration", function() {
...
+    cy.contains("Sign in with dex")
+      .click()
+
+    cy.dexStaticLogin()
+
+    cy.get("button[aria-label=\"Toggle menu\"]")
+      .click()
+
+    // Some requests are slow to start and need a second
+    cy.wait(1000)
```
Comment (Xartos) [inline tests/end-to-end/grafana-dev.cy.js:37]: There's no functionality in cypress to like "Wait until I see a thing containing X"?

**Review by Xartos — APPROVED** [review]
LGTM, with a small question

File: tests/end-to-end/grafana-dev.cy.js (hunk 1 above)
Comment (aarnq) [reply tests/end-to-end/grafana-dev.cy.js:37]: Yes, but the thing seem to be from their blog so it is not constant. I can see if that can be disabled instead.

Comment (aarnq) [reply tests/end-to-end/grafana-dev.cy.js]: I found a better wait condition.

**Comment by aarnq** [comment]
To run this locally to see what is actually tested ensure you have `bats`, `make` and `npm`, then go to `test/`, run `make dep`, then `npx cypress open` and you should get the Cypress UI were you can click through to E2E, with your favourite browser, and then run trough the specs.

**Pull request merged**